worker/instancepoller/worker.go

```diff
@@ -12,28 +12,80 @@ import (
 	"gopkg.in/juju/worker.v1"
 	"gopkg.in/juju/worker.v1/catacomb"
 
-	"github.com/juju/juju/api/instancepoller"
+	"github.com/juju/juju/apiserver/params"
 	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/core/status"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/context"
+	"github.com/juju/juju/environs/instances"
 	"github.com/juju/juju/worker/common"
 )
 
+//go:generate mockgen -package mocks -destination mocks/mocks_watcher.go github.com/juju/juju/core/watcher StringsWatcher
+//go:generate mockgen -package mocks -destination mocks/mocks_instances.go github.com/juju/juju/environs/instances Instance
+//go:generate mockgen -package mocks -destination mocks/mocks_cred_api.go github.com/juju/juju/worker/common CredentialAPI
+//go:generate mockgen -package mocks -destination mocks/mocks_instancepoller.go github.com/juju/juju/worker/instancepoller Environ,Machine
+
+// ShortPoll and LongPoll hold the polling intervals for the instance
+// updater. When a machine has no address or is not started, it will be
+// polled at ShortPoll intervals until it does, exponentially backing off
+// with an exponent of ShortPollBackoff until a maximum(ish) of LongPoll.
+//
+// When a machine has an address and is started LongPoll will be used to
+// check that the instance address or status has not changed.
+var (
+	ShortPoll        = 1 * time.Second
+	ShortPollBackoff = 2.0
```

Comment:
That is the same backoff used by the previous version. An improvement that I have made to the new implementation is that whenever the worker picks up a change in the machine doc it will immediately move the machine to the short poll group (if not already there) **and** reset its poll interval so we immediately query it.

We could either make the short-poll 1m or implement some sort of automatic reset (maybe with a bit of jitter) after hitting some hard-limit.